src/NetscriptFunctions/Formulas.ts

```diff
@@ -134,6 +135,12 @@ export function NetscriptFormulas(): InternalAPI<IFormulas> {
         checkFormulasAccess(ctx);
         return donationForRep(reputation, person);
       },
+      sharePower: (ctx) => (_threads, _cpuCores) => {
+        const threads = helpers.positiveInteger(ctx, "threads", _threads);
+        const cpuCores = helpers.positiveInteger(ctx, "cpuCores", _cpuCores);
+        checkFormulasAccess(ctx);
+        return calculateShareBonusWithAdditionalThreads(threads, cpuCores);
```

Comment:
This function uses the shared state `shareThreads`, and is thus not idempotent (it will give different answers at different times).